packages/website/docs/components/patterns/AI Writing Assistant/main.js

```diff
@@ -0,0 +1,218 @@
+	onMenuItemClick(e.detail.text);
+});
+
+
+document.addEventListener('keydown', function (event) {
+	const isMac = navigator.userAgent.includes('Mac')
+	if (event.key === 'Escape') {
+		stopLoading();
+	}
+
+	if (event.key === 'F4' && event.shiftKey) {
+		if (input.value.length) {
+			updateMenuVisibility(menuRegenerate, true, true);
+			updateMenuVisibility(menuGenerate, false);
+		} else {
+			updateMenuVisibility(menuGenerate, true, true);
+			updateMenuVisibility(menuRegenerate, false);
+		}
+	}
+
+	if (event.key === 'Enter' && (event.ctrlKey || (isMac && event.metaKey))) {
+		transitionToLoadState();
+	}
+
+	if (event.key === 'Z' && (event.ctrlKey || (isMac && event.metaKey)) && event.shiftKey) {
+		if (inputValues.length > 1 && currentInputIndex > 0) {
+			currentInputIndex--;
+			const previousValue = inputValues[currentInputIndex];
+			updateInputState(true, previousValue, 'Placeholder');
+		} else {
+			console.log("No previous values available.");
+		}
+	}
+	if (event.key === 'Y' && (event.ctrlKey || (isMac && event.metaKey)) && event.shiftKey) {
+		if (inputValues.length > 0 && currentInputIndex < inputValues.length - 1) {
+			currentInputIndex++;
+			const nextValue = inputValues[currentInputIndex];
+			updateInputState(true, nextValue, 'Placeholder');
+		} else {
+			console.log("No next values available.");
+		}
+	}
+});
+
+function onMenuItemClick(text) {
+	switch (text) {
+		case "Generate":
+			transitionToLoadState();
+			break;
+		case "Regenerate":
+			transitionToLoadState();
+			break;
+		case "Fix spelling and grammar":
+			transitionToLoadState();
+			break;
+		// Translate:
+		case "Bulgarian":
+			transitionToLoadState();
+			break;
+		case "English":
+			transitionToLoadState();
+			break;
+		case "German":
+			transitionToLoadState();
+			break;
+		default:
+	}
+}
+
+function stopLoading() {
+	clearTimeout(loadTimeout);
+	let lastInputValue = inputValues[inputValues.length - 1] || '';
+	currentInputIndex = inputValues.indexOf(lastInputValue);
+	updateInputState(true, lastInputValue, 'Placeholder');
+	input.classList.remove('loading-input');
+	updateIconState("ai", "AI Writing Assistant (Shift + F4)");
+}
+
+function updateIconVisibility(visible) {
+	if (visible) {
+		icon.classList.remove("hidden");
+		icon.classList.add("show");
+	} else {
+		icon.classList.remove("show");
+		icon.classList.add("hidden");
+	}
+}
+
+function updateIconState(iconName, iconAccessibleName) {
+	icon.name = iconName;
+	icon.accessibleName = iconAccessibleName;
+}
+
+function updateInputState(isEditable, value, placeholder) {
+	if (isEditable) {
+		input.removeAttribute('readonly');
+	} else {
+		input.setAttribute('readonly', true);
+	}
+	input.value = value;
+	input.placeholder = placeholder;
+}
+
+function addInputValue(value) {
+	if (value && value !== "Analyzing request...") {
+		inputValues.push(value);
+		currentInputIndex = inputValues.length - 1;
+		console.log("Current inputValues:", inputValues);
+	}
+}
```

Comment:
Same as above (regarding the versioning).

Reply:
Removed :)